tests/queries/0_stateless/03274_handle_is_deleted_with_prewhere_opt.sql

```diff
@@ -0,0 +1,24 @@
+drop table if exists 03274_prewhere_is_deleted;
+
+create table 03274_prewhere_is_deleted (number UInt64, value String, version UInt64, is_deleted UInt8) engine=ReplacingMergeTree(version,is_deleted) ORDER BY number AS SELECT number, toString(number), 1, 1 FROM numbers(10000000);
```

Comment:
Personally I would add a test with mixture of records with and without is_deleted